symphony/bdk/core/service/datafeed/abstract_datafeed_loop.py

```diff
@@ -97,12 +139,18 @@ async def handle_v4_event_list(self, events: List[V4Event]):
 
         :param events: the list of the received datafeed events.
         """
+        if not events:
+            return
+
         for event in filter(lambda e: e is not None, events):
             for listener in self.listeners:
                 if await listener.is_accepting_event(event, self.bdk_config.bot.username):
-                    await self._dispatch_on_event_type(listener, event)
+                    asyncio.create_task(self._dispatch_on_event_type(listener, event))
```

Comment:
do we have to do something in term of error handling? I remember that is the legacy BDK there were done callbacks used
if an exception is raised in the task it won't break the DF loop or anything?

Reply:
to support the soft/hard kill will we have to collect the tasks somewhere so we can cancel() all of the running ones/

Reply:
If an exception is raised inside the listener, then asyncio logs it for us:
```
2021-03-24 10:01:58,850 - root - DEBUG - iIaUeR - Received event: Ojwe2s5mMfJbHPlLbEZRUH___oech-ZubQ
2021-03-24 10:02:03,208 - asyncio - ERROR -  - Task exception was never retrieved
future: <Task finished name='Task-13' coro=<AbstractDatafeedLoop._dispatch_on_event_type() done, defined at /Users/elias.croze/workspace/IdeaProjects/symphony-api-client-python/symphony/bdk/core/service/datafeed/abstract_datafeed_loop.py:150> exception=ValueError('error inside listener')>
Traceback (most recent call last):
  File "/Users/elias.croze/workspace/IdeaProjects/symphony-api-client-python/symphony/bdk/core/service/datafeed/abstract_datafeed_loop.py", line 163, in _dispatch_on_event_type
    await listener_method(event.initiator, event_field)
  File "/Users/elias.croze/workspace/IdeaProjects/symphony-api-client-python/examples/datafeed.py", line 28, in on_message_sent
    raise ValueError("error inside listener")
ValueError: error inside listener
```
IMHO this is enough.

Reply:
To retrieve unfinished tasks, we can use: https://docs.python.org/3/library/asyncio-task.html#asyncio.all_tasks

Reply:
is there a risk we will retrieve user's define tasks and stop them (like if the bot is doing something else beside running the DF loop)?

Reply:
I added a list with the listener tasks. Please bear in mind lists are thread safe: https://stackoverflow.com/a/6319267